scripts/sorse_zenodo_upload.py

```diff
@@ -0,0 +1,212 @@
+            creator = dict()
+            creator['name'] = aut['name']
+            if not 'affiliation' in aut:
+                logging.error('Author %s in event %s does not have an affiliation! Defaulting to N/A', aut['name'], path)
+                creator['affiliation'] = 'N/A'
+            else:
+                aff_index = aut['affiliation']
+                # find affiliation string
+                for keyval in affiliations:
+                    if aff_index == keyval['index']:
+                        creator['affiliation'] = keyval['name']
+                        break
+            if 'orcid' in aut:
+                creator['orcid'] = aut['orcid'] 
+            
+            # do not add other author fields, because the Zenodo API will complain
+            creators.append(creator)
+        
+        # update .md with DOI
+        post['doi'] = doi
+        filename, file_extension = os.path.splitext(str(path))
+        outputpath = str(path) if overwrite else filename+'-new'+file_extension
+        output_file = open(outputpath, 'wb')
+        frontmatter.dump(post, output_file)
+        output_file.close()
+        # generate PDF
+        temp_file = open(workingpath + '/' + path.name, 'wb')
+        frontmatter.dump(post, temp_file)
+        temp_file.close()
+        # copyfile(outputpath, './generate-pdf/'+ path.name) # this might not finish before the subprocess call
+        subprocess.call(['sh', './generate-pdfs.sh'])
+        # os.remove('./generate-pdf/'+ path.name)
+
+        # The target URL is a combination of the bucket link with the desired filename
+        # seperated by a slash.
+        pdfpath = workingpath + '/' + os.path.basename(filename) + '.pdf' # use the event filename for the pdf
+        logging.info("Uploading file contents for %s", pdfpath)
+        
+        with open(pdfpath, "rb") as fp:
+            r = requests.put(
+                "%s/%s" % (bucket_url, os.path.basename(filename) + '.pdf'),
+                data=fp,
+                params=params,
+            )
+        if (r.status_code != 200):
+            logging.error("Failed upload file contents! Response: %i: %s", r.status_code, r.content)
+            print("Error processing {}, check log file for more information".format(path))
+            continue
+        logging.info("File contents uploaded for %s", pdfpath)
+    
+        # add metadata to deposition
+        data = { 'metadata': {
+            'publication_date': str(post['date']),
+            'title': post['title'],
+            'upload_type': 'publication',
+            'publication_type': 'conferencepaper',
+            'description': post.content,
+            'creators': creators,
+            'communities': [{'identifier': communityid}],
+            'conference_title': 'International Series of Online Research Software Events',
+            'conference_acronym': 'SORSE', 
+            'conference_url': 'https://sorse.github.io',
+            'access_right': 'open',

```

Comment:
It assumes all publishable content is `open`. Events in the _events folder that are not allowed to be published should have a 'publish: no' field in the frontmatter.